docs/source/API/core/MultiGPUSupport.rst

```diff
@@ -0,0 +1,149 @@
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    cudaGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    cudaStream_t stream;
+    cudaSetDevice(N);
+    cudaStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::Cuda exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    cudaSetDevice(N);
+    cudaStreamDestroy(stream);
+
+HIP
+~~~
+
+For the HIP backend, like with CUDA, the user creates a ``hipStream_t`` object and passes it to the ``Kokkos::HIP`` constructor.
+
+.. note:: The user is expected to manage the lifetime of the stream. The related execution space needs to be destroyed before destroying the stream.
+
+.. warning:: Multi-GPU only supported for ROCm 5.6 and later. Because of the lack of HIP API functions for querying a
+             stream's device before ROCm 5.6, constructing a ``Kokkos::HIP`` instance on a non-default device isn't
+             supported.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    hipGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    hipStream_t stream;
+    hipSetDevice(N);
+    hipStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::HIP exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    hipSetDevice(N);
+    hipStreamDestroy(stream);
+
+SYCL
+~~~~
+
+For the SYCL backend, the user creates a ``sycl::queue`` object and passes it to the ``Kokkos::SYCL`` constructor.
+
+.. code-block:: cpp
+
+    // Get list of devices available
+    std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+
+    // Choose some 0 <= N < gpu_devices.size()
+    int N = ...;
+
+    // Create a queue on device N.
+    // Note: Kokkos requires SYCL queues to be "in_order"
+    sycl::queue queue{gpu_devices[N], sycl::property::queue::in_order()};
+
+    // Create execution space
+    Kokkos::SYCL exec_space(queue);
+
+    // Use execution space
+    /* ... */
+
+Using Kokkos Methods
+--------------------
+
+Once an execution space has been created on the chosen device, the execution space must be passed to all Kokkos methods
+intended to be used on the chosen device. If no execution space is passed, Kokkos will use `DefaultExecutionSpace`.
+
+Allocating Managed Views
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To allocate a managed view on device, pass the execution space to ``Kokkos::view_alloc()``.
+
+Example:
+
+.. code-block:: cpp
+
+    Kokkos::View<int*> V(Kokkos::view_alloc("V", exec_space), 10);
```

Comment:
I don't agree with removing the `ExecutionSpace` here.
```suggestion
    ExecutionSpace exec_space(...);
    Kokkos::View<int*, typename ExecutionSpace::memory_space> V(Kokkos::view_alloc("V", exec_space), 10);
```

Reply:
can you explain why?

Reply:
It matters what the type of the execution space is. It has to match with the memory space. I'm fine with using the defualt one here for it to read.
```suggestion
    Kokkos::DefaultExecutionSpace exec_space(...);
    Kokkos::View<int*> V(Kokkos::view_alloc("V", exec_space), 10);
```
but still prefer using a generic one.

Reply:
I agree, that spelling the executionspace out in the example is a good idea. But don't we want to imply that these Views should be created with the exec space that is associated to a stream? This exec space was called `exec_space` in the snippets above.
Does this make sense?

Reply:
I'm not arguing about the spelling of `exec_space` and we can be more explicit by adding a comment
```suggestion
    ExecutionSpace exec_space(...); // constructed as above
    Kokkos::View<int*, typename ExecutionSpace::memory_space> V(Kokkos::view_alloc("V", exec_space), 10);
```

Reply:
Updated by querying the type from execution space. I agree with Jakob, I was assuming the reader would follow `exec_space` through the doc.